web/pages/api/login.ts

```diff
@@ -0,0 +1,116 @@
+import {
+  errorResponse,
+  errorNotAllowed,
+  errorRequiredAttribute,
+} from "api-helpers/errors";
+
+import { gql } from "@apollo/client";
+import { CredentialType, NextApiRequestWithBody } from "types";
+import { getAPIServiceClient } from "api-helpers/graphql";
+import { generateUserJWT, generateUserTempJWT } from "api-helpers/utils";
+import { NextApiResponse } from "next";
+import { verifyProof } from "api-helpers/verify";
+import dayjs from "dayjs";
+import { internal as IDKitInternal } from "@worldcoin/idkit";
+
+export type LoginRequestBody = {
+  proof?: string;
+  nullifier_hash?: string;
+  merkle_root?: string;
+  signal_type?: CredentialType;
+};
+
+export type LoginResponse =
+  | {
+      tempToken: string;
+      token?: never;
+    }
+  | {
+      tempToken?: never;
+      token: string;
+    };
+
+const query = gql`
+  query FindUserByNullifier($nullifier_hash: String) {
+    user(where: { world_id_nullifier: { _eq: $nullifier_hash } }) {
+      id
+      team_id
+      world_id_nullifier
+      email
+    }
+  }
+`;
+
+export default async function login(
+  req: NextApiRequestWithBody<LoginRequestBody>,
+  res: NextApiResponse<LoginResponse>
+) {
+  if (!req.method || !["POST", "OPTIONS"].includes(req.method)) {
+    return errorNotAllowed(req.method, res);
+  }
+
+  const { proof, nullifier_hash, merkle_root, signal_type } = req.body;
+
+  const invalidBody = !proof || !nullifier_hash || !merkle_root || !signal_type;
```

Comment:
@paolodamico I've made it like this because it seems like typescript not recognizing if you check params using an array of strings and some cycle. 
I've made an [example](https://www.typescriptlang.org/play?target=9#code/PTAEEEDkGEAkHkBKAuUBRAHgQwLYAcAbAU1AHcBLAFwAtQtQBjATweMeqIYGsAoBgewB2AZ0qhqWQQBNiAJ1ABeUAAoAlIoB8oAN49Q+xkNGgARvylNU20ADN+-APypRs8oIDmAGlNZZT0C5u7qAAvoo6tvaoAOQAjABMAMzRoTx6BgIiYtZ2-N4mvqHhZhZpBpHyypnGWJSU8vw2oADa6eX6AES5HZ5t7R0Fsh19ALp0wi1d9j2gA74dI+q67QbkTcoAhCVMzbX1izp9K-qyRJQArrKChiL8xAB0RLKy-LLK0U8vstGqRwYhfQBfRAoAAktcaOQJgwsMISB0AGLTOjSWYAIXmoChAUo5AIBFA0UCHlAAB9QOdpEQbG4iFJovdQNAONw6GYAG4kQT8MRYGw2TiUOmgISgShMPBEYQMVx4Sh9apiJH8cK5BVGMQY+RKQZlAynC5XCLK7xa1JAkEAWmtNttdvtDrtaRBUDgSFQaCwDFoeF8uHYnF4ipRPI4slgkhkT3Cak0h3Kwe2Vkijmc9SC+V8-mJwTCShyUUJCWSqQTGoiuUz8jzpnMTD1+mDbnZWAI5CkaLr4Q2uTJ5K2vgbWPWzdb7c7FiWfxOZ0u12qdyIj2er3en1eP0BQ5B4LF1GxMLhs2VHRRUnRmOxQgITEJOYZoAAKhKpTLyHLxLDQKcBO5BOQAC9hW9QN7nVLJQGVVV7HA4wzR1Qc+gNOdjXsU1CgBEIgA), please, take a look.